### ISSUE TITLE
createTableIfMissing not working

### ISSUE THREAD

**Issue by vogler** [voxpelli/node-connect-pg-simple#200]
With the following
~~~typescript
import connectPgSimple from 'connect-pg-simple';
const store = new (connectPgSimple(session))({createTableIfMissing: true} as any);
app.use(session({secret: '...', saveUninitialized: true, resave: false, store})); // defaults: httpOnly
~~~
I just get `Failed to prune sessions: relation "session" does not exist` on start and `error: relation "session" does not exist` on the first request.

Also, without `as any` I get `Argument of type '{ createTableIfMissing: boolean; }' is not assignable to parameter of type 'PGStoreOptions'.`, so `@types/connect-pg-simple` seems to be outdated.

~~~json
    "@types/connect-pg-simple": "^4.2.2",
    "connect-pg-simple": "^6.2.1",
    "express": "^4.17.1",
    "express-session": "^1.17.1",
    "typescript": "^4.1.5",
~~~

**Comment by voxpelli** [comment]
The `@types/connect-pg-simple` should not be used, `connect-pg-simple` comes with its own types, I should look into getting that one deprecated.

Regarding main issue: I'll give it a look

**Comment by voxpelli** [comment]
Sorry, I misspoke earlier, this is the one of my modules that actually doesn't yet generate its own types, see #155

I unfortunately also got sidetracked with an issue in another project that affected this project, so never got around to look at this in the weekend, sorry :/

**Comment by vogler** [comment]
No problem, I'm using https://github.com/kleydon/prisma-session-store - since I'm using Prisma anyway this is a better fit.

**Comment by prionkor** [comment]
Hello, I am having the same issue. Table is not created if I set `createTableIfMissing` as `true`

Code sample:

```
const PgSession = pgSession(session);
const pool = new Pool({
  host: process.env.DB_HOST,
  user: process.env.DB_USER,
  password: process.env.DB_PASS,
  database: process.env.DB_NAME,
});

const store = new PgSession({
  createTableIfMissing: true,
  tableName: 'UserSessions',
  pool,
});

```

**Comment by prionkor** [comment]
This bug makes the module unusable for my use cases. So I have to move to a new module. https://www.npmjs.com/package/connect-session-sequelize

**Comment by voxpelli** [comment]
@prionkor You can always set up the tables yourself, as shown in: https://github.com/voxpelli/node-connect-pg-simple#installation

The `createTableIfMissing` is a helper for those who can't.

I haven't got that much to go on here and I'm quite short on time at the moment so I would have to debug it deeper, and can't promise when I get time to do that.

But for future reference, for those who might arrive here: If  `createTableIfMissing` doesn't work, just go with the default behavior of seting up the table yourself.

**Comment by lf94** [comment]
I too am hitting this bug. It just won't create itself.

**Comment by lf94** [comment]
> The @types/connect-pg-simple should not be used, connect-pg-simple comes with its own types, I should look into getting that one deprecated.

I don't see its types declared anywhere? How are you using its types? (Looking at it, tsconfig points to index.js, and there is lots of annotations)

**Comment by lf94** [comment]
I suggest anyone coming to the repo try out https://www.npmjs.com/package/connect-session-sequelize instead, which is much more maintained. Nothing against voxpelli, they have done excellent work with connect-pg-simple. connect-pg-simple works but does not fit my need of TypeScript + auto creating the tables.

**Comment by voxpelli** [comment]
> I don't see its types declared anywhere? How are you using its types? (Looking at it, tsconfig points to index.js, and there is lots of annotations)

I corrected myself earlier:

> Sorry, I misspoke earlier, this is the one of my modules that actually doesn't yet generate its own types, see #155

Part of the reason is that this has to interact with express-session and for a long time those types were a bit so-so, making it hard to make something useful of it.

Apart from that: The focus of this module is to be the simplest postgres session module for express, thus depending on nothing but the pg module itself, enabling it to be used for most scenarios.

If one needs a more complex solution, then it’s correct to look for that in another solution.

If one believes this module fails in delivering what it intends to deliver, then let’s fix it 🙂

**Comment by lf94** [comment]
Well said :slightly_smiling_face: I want to heavily emphasize I am not implying anything negative about the work - it is once again, excellent. I would try to fix the mentioned issues myself, I just don't have the time either right now. :laughing:

**Comment by vladtabachuk** [comment]
Hello  Mr. voxpelli same issue with me, please can you tell this CREATE TABLE command to extcute it in postgreSql itself?

**Comment by aboqasem** [comment]
Make sure you install `connect-pg-simple@next` everyone!

**Comment by voxpelli** [comment]
Thanks @aboqasem, I had forgotten that `7.0.0` was still in a prerelease state, I have now released a stable `7.0.0`.

Sorry everyone for not thinking of this myself earlier. I'll close this issue as solved for now, if an update to `7.0.0` does not solve it, then comment and I can reopen.

**Comment by CharaHarimata** [comment]
I've come across the same error as specified prior:
```shell
Failed to prune sessions: relation "sessions" does not exist
node:events:368
      throw er; // Unhandled 'error' event
      ^
```

I currently remain unable to implement database storage because of this.

**Comment by isaacgr** [comment]
> Thanks @aboqasem, I had forgotten that `7.0.0` was still in a prerelease state, I have now released a stable `7.0.0`.
> 
> Sorry everyone for not thinking of this myself earlier. I'll close this issue as solved for now, if an update to `7.0.0` does not solve it, then comment and I can reopen.

I'm seeing this with 7.0.0. Once created manually theres no issue.
